core/src/main/scala/org/apache/spark/sql/SnappyContext.scala

```diff
@@ -544,41 +543,108 @@ class SnappyContext protected[spark](
     }
   }
 
-
   /**
-   * Create Index on a SnappyData table (created by a call to createTable).
-   * @todo how can the user invoke this? sql?
-   */
-  private[sql] def createIndexOnTable(tableIdent: QualifiedTableName,
-      sql: String): Unit = {
-    //println("create-index" + " tablename=" + tableName    + " ,sql=" + sql)
+    * Create an index on a table.
+    * @param indexName Index name which goes in the catalog
+    * @param baseTable Fully qualified name of table on which the index is created.
+    * @param indexColumns Columns on which the index has to be created
+    * @param options Options for indexes. For e.g.
+    *                column table index - ("COLOCATE_WITH"->"CUSTOMER").
+    *                row table index - ("INDEX_TYPE"->"GLOBAL HASH") or ("INDEX_TYPE"->"UNIQUE")
+    */
+  def createIndexOnTable(indexName: String,
+                         baseTable: String,
+                         indexColumns: Seq[String],
```

Comment:
Please add a java compatible version to API as well. Each time one adds an API , he/she should also think about how it can be accessed by Java code. Seq[String] can be List<String> and Map can be java.util.Map<String.String>